### ISSUE TITLE
Saving bai index

### ISSUE THREAD

**Issue by bkehr** [seqan/seqan#1081]
Hi,

I have been using the `_saveIndex()` function for a `BamIndex<Bai>` although it is currently only called in a function that starts with (bam_io/bam_index_bai.h, line 614):
```
SEQAN_FAIL("This does not work yet!");
```

The `_saveIndex()` function is almost working fine for me, only line 573
```
__int32 numIntervals = length(index._linearIndices);
```
needs to be fixed to
```
__int32 numIntervals = length(linearIndex);
```
.

Cheers, Birte

**Comment by esiragusa** [comment]
@holtgrewe Hi, do you know if this works or not?

**Comment by esiragusa** [comment]
@weese Hi, do you know if this works or not?

**Comment by esiragusa** [comment]
Hey @bkehr! I know that Dave worked on it, but don't know if it works now. Did you call ```_saveIndex()``` directly or through ```buildIndex()```?

**Comment by bkehr** [comment]
@esiragusa Hi Enrico. I called `_saveIndex()` directly and then it works with the fix. I didn't call it through `buildIndex()`.

**Comment by esiragusa** [comment]
Ah, did you open and then save an index created with samtools? Cheers!

**Comment by bkehr** [comment]
Yes, I opened a Bai file that I created with samtools, saved it with the fixed function, used the new bai file with samtools, ... It all worked.

**Comment by esiragusa** [comment]
All right. Thanks!

**Comment by esiragusa** [comment]
Fixed in #1201, #1202.

@bkehr Now you can use:
* ```open(bai, finename)```
* ```build(bai)``` (which it's still broken but will be hopefully fixed in another PR by @weese)
* ```save(bai, finename)```

Cheers!
